code/game/objects/barricades.dm

```diff
@@ -0,0 +1,379 @@
+	code = new_code
+	frequency = new_frequency
+	if(frequency)
+		SSradio.add_object(src, frequency)
+
+
+/obj/structure/barrier/four_way/attackby(obj/item/I, mob/user)
+	if(istype(I, /obj/item/device/assembly/signaler))
+		var/obj/item/device/assembly/signaler/signaler = I
+		if((signaler.code == code) && (signaler.frequency == frequency)) // Linked signaler is used, break the link
+			update_wifi_password()
+			signaler.audible_message("\icon[signaler] *beep*")
+		else if(!frequency || !code) // There is no link, establish one
+			update_wifi_password(signaler.code, signaler.frequency)
+			signaler.audible_message("\icon[signaler] *beep* *beep*")
+		else // Access denied
+			signaler.audible_message("\icon[signaler] *beep* *beep* *beep*") // Silly and cryptic, yet authentic signaler feedback
+		return
+
+	var/list/usable_qualities = list(QUALITY_BOLT_TURNING)
+	if(health < maxHealth)
+		usable_qualities.Add(QUALITY_WELDING)
+	if(code || frequency)
+		usable_qualities.Add(QUALITY_PULSING)
+
+	var/tool_type = I.get_tool_type(user, usable_qualities)
+
+	switch(tool_type)
+		if(QUALITY_WELDING)
+			if(health < maxHealth)
+				if(I.use_tool(user, src, WORKTIME_NORMAL, tool_type, FAILCHANCE_NORMAL,  required_stat = STAT_MEC))
+					user.visible_message(SPAN_NOTICE("\The [user] repairs \the [src]."), SPAN_NOTICE("You repair \the [src]."))
+					health = maxHealth
+					is_damaged = FALSE
+					update_icon()
+					return
+		if(QUALITY_BOLT_TURNING) // Deconstruction and hacking be harder when barrier is deployed
+			if(I.use_tool(user, src, (is_open ? WORKTIME_LONG : WORKTIME_NORMAL), tool_type, (is_open ? FAILCHANCE_CHALLENGING : FAILCHANCE_NORMAL),  required_stat = STAT_MEC))
+				for(var/material_name in matter)
+					var/material/material_datum = get_material_by_name(material_name)
+					material_datum.place_sheet(loc, amount = matter[material_name])
+				user.visible_message(SPAN_NOTICE("\The [user] dismantles \the [src]."), SPAN_NOTICE("You dismantle \the [src]."))
+				qdel(src)
+				return
+		if(QUALITY_PULSING)
+			if(I.use_tool(user, src, (is_open ? WORKTIME_LONG : WORKTIME_NORMAL), tool_type, (is_open ? FAILCHANCE_CHALLENGING : FAILCHANCE_NORMAL),  required_stat = STAT_MEC))
+				update_wifi_password()
+				return
+
+	user.do_attack_animation(src)
+	if(I.hitsound)
+		playsound(loc, I.hitsound, 50, 1, -1)
+	visible_message(SPAN_DANGER("[src] has been hit by [user] with [I]."))
+	user.setClickCooldown(DEFAULT_ATTACK_COOLDOWN)
+	take_damage(I.force * I.structure_damage_factor)
+
+
+/obj/structure/barrier/four_way/receive_signal(datum/signal/signal, receive_method, receive_param)
+	ASSERT(signal)
+	if(signal.encryption == code)
+		toggle_barrier()
+
+
+/obj/structure/barrier/four_way/proc/toggle_barrier()
+	is_open = !is_open
+	density = !density
+	playsound(loc, 'sound/machines/Custom_bolts.ogg', 20)
+	update_icon()
+	flick(icon_state + "_animation", src)
+
+
+/obj/structure/barrier/hedgehog
+	name = "Serb hedgehog"
```

Comment:
```suggestion
	name = "serbian hedgehog"
```
"serb" is only used in code, not in any names
Also avoid uppercases

Reply:
It meant to be used in the name though.
That thing normally called Czech hedgehog, because it originates from Czechoslovakia.
Ain't no space Czechoslovakia in Eris lore, but we have space Serbia of sorts. Well, space Serbians at least.
So this is called after Serbia instead, and country/nation/faction name should normally be capitalized.

Reply:
I agree and understand that this is a word-play on Czech hedgehog, but "Serb" isn't a real adjective. Things that you see a lot, not disks or interfaces tend to be lowercased as they tend to be eyesores otherwise.
![изображение](https://github.com/discordia-space/CEV-Eris/assets/57810301/f4bf307b-bd6a-4a2f-9f22-e586973f42cf)
![изображение](https://github.com/discordia-space/CEV-Eris/assets/57810301/2f56cddb-c9c1-4b57-acf0-21dce3c2d426)